Makefile

```
@@ -0,0 +1,77 @@
+# Enable running all commands in the same subshell
+.ONESHELL:
+
+# Define the name of the virtual environment
+VENV := venv
+
+# Detect the operating system (Windows vs Unix-like)
+OSFLAG :=
+ifeq ($(OS),Windows_NT)
+    OSFLAG := windows
+else
+    UNAME_S := $(shell uname -s)
+    ifeq ($(UNAME_S),Linux)
+        OSFLAG := linux
+    endif
+    ifeq ($(UNAME_S),Darwin)
+        OSFLAG := macos
+    endif
+endif
+
+# Default target: setup the environment, install dependencies, open VSCode, and show activation instructions
+.PHONY: start
+start: venv install open_vscode print_activation_instructions
+
+# Create the virtual environment only if it doesn't exist
+.PHONY: venv
+venv:
+ifeq ($(OSFLAG),windows)
+	if not exist $(VENV)\Scripts\activate (
+		python -m venv $(VENV)
+	)
+else
+	if [ ! -f "$(VENV)/bin/activate" ]; then
+		python3 -m venv $(VENV)
+	fi
```

```suggestion
	if [ ! -f "$(VENV)/bin/activate" ]; then python3 -m venv $(VENV); fi
```
Apparently a missing semicolon, and my system doesn't like it when it is in multiple lines.

---

I think its better if you test it on your system and push a solution, its more or less impossible for me to fix it, as I only have access to Windows and Linux, altough doing it for Linux would require me to set everything up as I havent used Linux for a while.

---

Please accept my suggestion, as I have resolved the issue in my system.

---

Great, but I cant seem to find it, did you push it to my branch or where can I find it?

---

It also seems like I forgot to push the changes I made yesterday in an attempt to fix the issue.

---

@simondereuver It's now works with the new changes you made